### PR TITLE
Add gometalinter and Dockerfile 

### PR DESCRIPTION
**Pull request by gpaul**
This PR adds the `make check`, `make gofmt` and `make shell` commands:

`make check` runs various linters using the `gometalinter` over the project sources.
`make gofmt` runs `gofmt -w -s` on all the project files.
`make shell` drops the caller into a docker shell, ready to build the project.

The `make build` target has been modified to use docker unless `make build DOCKER=no` is specified.

This work paves the way for setting up CI.

Fixes https://jira.mesosphere.com/browse/DCOS_OSS-1922

### CODE REVIEW

**Comment by gpaul** [comment]
Test this please

**Comment by gpaul** [comment]
Test this please

**Comment by gpaul** [comment]
Test this please

**Comment by gpaul** [comment]
Test this please

File: gometalinter.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (jdef) [inline gometalinter.conf]: :+1: is this type of configuration a new ability for gometalinter?

**Comment by jdef** [comment]
maybe add a build badge to the readme? as per https://jenkins.mesosphere.com/service/jenkins/job/public-csilvm/badge/

File: gometalinter.conf (hunk 1 above)
Comment (gpaul) [reply gometalinter.conf]: Actually no idea! I just saw it today and assumed it has always been there

**Comment by gpaul** [comment]
Build badge added, thanks for the tip

**Comment by gpaul** [comment]
Test this please

**Pull request merged**